raiden/network/transport/matrix.py

```diff
@@ -662,58 +737,58 @@ def _get_room_for_address(
+                        error=error,
                     )
-                except MatrixRequestError as error:
-                    if error.code == 409:
-                        message = 'seems to have been created by peer meanwhile.'
-                    else:
-                        message = f'{error.code} {error.content}'
-                    self.log.info(f'Error creating room {room_name}: {message}. '
-                                  f'Retrying to join...')
-                    room_not_found = False
                 else:
-                    self.log.info('Room created successfully', room=room, invitees=invitees)
-                    return room
+                    self.log.info(f'Error joining room {room_name}: '
+                                  f'{error.content} {error.code}')
             else:
-                try:
-                    room = self._client.join_room(room_name_full)
-                except MatrixRequestError as error:
-                    if error.code == 404:
-                        self.log.info(
-                            f'Room {room_name_full} not found, trying to create it.',
-                            error=error,
-                        )
-                        room_not_found = True
-                    else:
-                        self.log.info(f'Error joining room {room_name}: '
-                                      f'{error.content} {error.code}')
+                self.log.info('Room joined successfully', room=room)
+                break
+
+            # if can't, try creating it
+            try:
+                room = self._client.create_room(
+                    room_name,
+                    invitees=invitees_uids,
+                    is_public=True,  # FIXME: debug only
+                )
+            except MatrixRequestError as error:
+                if error.code == 409:
+                    message = 'seems to have been created by peer meanwhile.'
                 else:
-                    self.log.info('Room joined successfully', room=room)
-                    return room
-            gevent.sleep(self._login_retry_wait)
-
-        room = self._client.create_room(
-            None,
-            invitees=[user.user_id for user in invitees],
-            is_public=True,  # FIXME: debug only
-        )
-        log.warning(
-            'Could not create nor join a named room. Successfuly created an unnamed one',
-            room=room,
-            invitees=invitees,
-        )
+                    message = f'{error.code} {error.content}'
+                self.log.info(f'Error creating room {room_name}: {message}. '
+                              f'Retrying to join...')
+            else:
+                self.log.info('Room created successfully', room=room, invitees=invitees)
+                break
+        else:
+            # if can't join nor create, create an unnamed one
+            room = self._client.create_room(
```

Comment:
Can't a `MatrixRequestError` exception be thrown from there too?

Reply:
No, unnamed rooms always succeed, if it raises, it's a bug and shouldn't be handled